modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -33,19 +37,17 @@ def do_remove(name, type)
       nil
     end
 
-    def execute(cmd, msg = nil, error_only = false)
+    def execute(*args, **keyword_args)
       tsecs = 5
       response = nil
       interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
-      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server} #{cmd}"
-
       std_in = std_out = std_err = nil
       begin
         timeout(tsecs) do
-          logger.debug "executing: #{command}"
-          std_in, std_out, std_err = Open3.popen3(command)
-          response = std_out.readlines
-          response += std_err.readlines
+          logger.debug { "executing: #{interpreter} #{args.join(" ")}" }
+          std_in, std_out, std_err = popen3(interpreter, *args)
```

Comment:
Here is a bug, previously it was: `cmd.exe` calls `dnscmd.exe` + arguments.

Now it is: `cmd.exe` + arguments that will not be understood by this program.

How it should be: `dnscmd.exe` + arguments:

```suggestion
          std_in, std_out, std_err = popen3('c:\Windows\System32\dnscmd.exe', @server, *args)
```

You can completely drop helper interpreter string, I suggest you create a helper array called `real_cmd` and then put `dnscmd.exe` + `@server` into it and append the `cmd` argument. That you can pass into `popen3` as well as into logging methods.